packages/merge-styles/src/styleToClassName.ts

```diff
@@ -201,8 +207,10 @@ export function applyRegistration(
           return '';
         });
 
-        // Insert.
-        stylesheet.insertRule(`${selector}{${rules}}`);
+        // Insert. Note if a media query, we must close the query with a final bracket.
+        const processedRule = `${selector}{${rules}}${(selector.indexOf('@media') === 0) ? '}' : ''}`;
```

Comment:
Maybe we should extract this line here into another function and UT it? It took me a bit to understand what was going on.